components/tyk-enterprise-portal/templates/statefulset-enterprise-portal.yaml

```diff
@@ -0,0 +1,235 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "tyk-enterprise-portal.fullname" . }}
+  labels:
+    app: {{ include "tyk-enterprise-portal.fullname" . }}
+    chart: {{ include "tyk-enterprise-portal.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  serviceName: service-{{ include "tyk-enterprise-portal.fullname" . }}
+  replicas: {{ .Values.replicaCount }}
+  updateStrategy:
+    # indicate which strategy we want for rolling update
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: {{ include "tyk-enterprise-portal.fullname" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ include "tyk-enterprise-portal.fullname" . }}
+        release: {{ .Release.Name }}
+      annotations:
+        checksum/secrets: "{{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}"
+        {{- if .Values.mounts }}
+        {{- range $secret := .Values.mounts }}
+        "checksum/secret-{{ $secret.name }}": {{ include (print "mounts/%s" $secret.filename) . | sha256sum }}
+        {{- end }}
+        {{- end }}
+    spec:
+{{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+{{- end }}
+{{- if .Values.tolerations }}
+      tolerations:
+{{ toYaml .Values.tolerations | indent 8 }}
+{{- end }}
+{{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+{{- end }}
+      containers:
+      - image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        name: enterprise-portal-{{ .Chart.Name }}
+        securityContext:
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          privileged: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+              - all
+        env:
+          - name: PORTAL_LICENSEKEY
+            valueFrom:
+              secretKeyRef:
+                {{ if .Values.secrets.useSecretName }}
+                name: {{ .Values.secrets.useSecretName }}
+                {{ else }}
+                name: secrets-{{ include "tyk-enterprise-portal.fullname" . }}
+                {{ end }}
+
+                key: EnterprisePortalLicense
```

Comment:
Can't we make this field configurable?

Reply:
Why should we? It's something documented and would just add unnecessary clutter to the yaml